src/main/java/net/dv8tion/jda/client/entities/impl/ApplicationImpl.java

```diff
@@ -0,0 +1,409 @@
+    {
+        this.api = api;
+
+        this.updateFromJson(object);
+    }
+
+    @Override
+    public RestAction<Application.Bot> createBot()
+    {
+        if (this.hasBot())
+            return new RestAction.EmptyRestAction<>(this.bot);
+
+        return new RestAction<Application.Bot>(this.api, Route.Applications.CREATE_BOT.compile(this.id), null)
+        {
+            @Override
+            protected void handleResponse(final Response response, final Request request)
+            {
+                if (response.isOk())
+                    request.onSuccess(ApplicationImpl.this.bot = new BotImpl(response.getObject()));
+                else
+                    request.onFailure(response);
+            }
+        };
+
+    }
+
+    @Override
+    public RestAction<Void> delete()
+    {
+        return new RestAction<Void>(this.api, Route.Applications.DELETE_APPLICATION.compile(this.id), null)
+        {
+            @Override
+            protected void handleResponse(final Response response, final Request request)
+            {
+                if (response.isOk())
+                    request.onSuccess(Void.TYPE);
+                else
+                    request.onFailure(response);
+            }
+        };
+    }
+
+    @Override
+    public final boolean doesBotRequireCodeGrant()
+    {
+        return this.doesBotRequireCodeGrant;
+    }
+
+    @Override
+    public boolean equals(final Object obj)
+    {
+        return obj instanceof ApplicationImpl && this.id.equals(((ApplicationImpl) obj).id);
+    }
+
+    @Override
+    public Bot getBot()
+    {
+        return this.bot;
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return this.description;
+    }
+
+    @Override
+    public final int getFlags()
+    {
+        return this.flags;
+    }
+
+    @Override
+    public String getIconId()
+    {
+        return this.iconId;
+    }
+
+    @Override
+    public String getIconUrl()
+    {
+        return this.iconId == null ? null
+                : "https://cdn.discordapp.com/app-icons/" + this.id + '/' + this.iconId + ".jpg";
+    }
+
+    @Override
+    public String getId()
+    {
+        return this.id;
+    }
+
+    @Override
+    public JDA getJDA()
+    {
+        return this.api;
+    }
+
+    @Override
+    public synchronized ApplicationManager getManager()
+    {
+        ApplicationManager m = this.manager;
+
+        if (m == null)
+            m = this.manager = new ApplicationManager(this);
+
+        return m;
+    }
+
+    @Override
+    public synchronized ApplicationManagerUpdatable getManagerUpdatable()
+    {
+        ApplicationManagerUpdatable m = this.managerUpdatable;
+
+        if (m == null)
+            m = this.managerUpdatable = new ApplicationManagerUpdatable(this);
+
+        return m;
+    }
+
+    @Override
+    public String getName()
+    {
+        return this.name;
+    }
+
+    @Override
+    public final List<String> getRedirectUris()
```

Comment:
leftover from accidentially having the final option checked in the getter generator